drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_util.c

```diff
@@ -919,6 +939,50 @@ static int nrf_wifi_util_trigger_rpu_recovery(const struct shell *shell,
 }
 #endif /* CONFIG_NRF_WIFI_RPU_RECOVERY */
 
+static int nrf_wifi_util_set_ps_data_retrival_info(const struct shell *shell, size_t argc,
+						   const char *argv[])
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	char *ptr = NULL;
+	unsigned long ps_data_retrieval_mech = 0;
+	unsigned long ps_data_retrieval_frame_interval_ms = 0;
+
+	ps_data_retrieval_mech = strtoul(argv[1], &ptr, 10);
+
+	if ((ps_data_retrieval_mech < 0) || (ps_data_retrieval_mech > 1)) {
+		shell_fprintf(shell,
+			      SHELL_ERROR,
+			      "Invalid value(%lu).\n", ps_data_retrieval_mech);
+		shell_help(shell);
+		return -ENOEXEC;
+	}
+
+	if (ps_data_retrieval_mech == 1) {
+		if ((argc == 3) && argv[2]) {
+			ps_data_retrieval_frame_interval_ms = strtoul(argv[2], &ptr, 10);
+		}
+	}
+
+	status = nrf_wifi_fmac_set_ps_data_retrieval_info(ctx->rpu_ctx,
+							  ps_data_retrieval_mech,
+							  ps_data_retrieval_frame_interval_ms);
+	if (status != NRF_WIFI_STATUS_SUCCESS) {
+		shell_fprintf(shell,
+			      SHELL_ERROR,
+			      "Programming data retrival info failed\n");
+		return -ENOEXEC;
+	}
+
+	ctx->conf_params.ps_data_retrieval_mech = ps_data_retrieval_mech;
```

Comment:
What is the purpose of storing this in a context if this is being programmed in the real-time to the FW?

Reply:
The user will be able to see current configured mechanism and interval using "wifi_util show_config".